### PR TITLE
Merge openfsharp 1

### PR DESCRIPTION
**Pull request by enricosada**
- e01e100 merge fix test, now is crossplatform (see commit message)
- 655df3d cleanup style for better diff and merge of some changes

### CODE REVIEW

**Comment by KevinRansom** [comment]
Thanks for taking care of this Enrico, it is a valuable thing to do. The changes look good except  for the additional try/with, I would like to understand why it was necessary in the open codebase before we take it.

Thanks

Kevin

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs [hunk 1]
```diff
@@ -104,7 +104,7 @@ type LanguagePrimitivesModule() =
         
         // reference type
         let resultRef = LanguagePrimitives.GenericComparison "ABC" "ABCDE"
-        Assert.AreEqual(resultRef,-68)
+        Assert.AreEqual(sign resultRef,-1)
```
Comment (latkin) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs]: Hmm, not a blocker, but the nunit APIs are authored to expect `Assert.AreEqual(expected, actual)` , whereas these are called the other way `Assert.AreEqual(actual, expected)`.  The failure message will be a little bit confusing.

Doesn't really matter, they were already wrong. In the future for new tests we can try to remember this though.

**Comment by latkin** [comment]
Agree with @KevinRansom. Looks great, but would want to know the origin of the exception handling.

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs (hunk 1 above)
Comment (forki) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs]: let shouldEqual (expected:'a) (actual:'a) = Assert.AreEqual(expexted,actual)

Is used in many projects and also enforces the correct type on compile time

Comment (forki) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs]: (the benefit is in using |>)

**Comment by dsyme** [comment]
IIRC it was because the simulated registry is not available on Mono in all circumstances and attempts to access it can fail.

So I don't really think the change should be brought into VF# as is. While changes like this (which myself and others put in) served a purpose a few years ago under a  "get it running on Mono/Linux/Mac at all costs" agenda, they don't seem appropriate now, especially since we're very likely to have to add even another version of cross-platformness to this repo as we consider bringing the F# compiler up on cross-platform editions of .NET Core.

So I'd vote for keeping this one out for now.

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs (hunk 1 above)
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/PrimTypes.fs]: good catch! i'll fix it, easy
A long time ago I was filling a bug report to NUnit about this, only to find the parameters were reversed :sweat_smile:

**Comment by enricosada** [comment]
yes, i was thinking that was ok because the current code do pretty much the same ( see https://github.com/Microsoft/visualfsharp/blob/fac44095a8dbc7303efa9866e0e2d9dddbdeae5d/src/utils/CompilerLocationUtils.fs#L79-L85 ). 

i'll remove the `try`, this code need a refactoring with a good `List.tryPick` and a list of options :smile: 

next time i'll separate whitespace cleanup and change of behavior.

**Comment by enricosada** [comment]
removed `try`
the commit 172c363 align to openfsharp before fix of tests with d385f25 for easier merge

File: src/utils/CompilerLocationUtils.fs [hunk 2]
```diff
@@ -202,11 +202,14 @@ module internal FSharpEnvironment =
         // Check for an app.config setting to redirect the default compiler location
         // Like fsharp-compiler-location
         try 
+            // FSharp.Compiler support setting an appkey for compiler location. I've never seen this used.
```
Comment (latkin) [inline src/utils/CompilerLocationUtils.fs]: It's used [in this repo](https://github.com/Microsoft/visualfsharp/blob/fsharp4/vsintegration/src/unittests/Unittests.dll.config#L5)

**Comment by latkin** [comment]
Looks ok to me.

**Comment by latkin** [comment]
This is applied

**Pull request closed without merging**